### PR TITLE
Feature/disable logging in tests sol 2786

### PR DESCRIPTION
**Pull request by ieva-improbable**
This PR makes sure that logging is done through `slf4j` logging facade instead of `System.out` where possible. It also disables logging for tests by adding an extra `log4j` logger configuration (configuration for tests specifies logging only error level logs, while the configuration for prod sets logging to debug level), as well disabling loading bar when tests are ran from gradle.

Up for debate:
* I didn't change logging in the code used for doc snippets as I thought it would be confusing for the readers. So these are still using `System.out.println` (for instance, _WetGrass.java_);
* Some tests reside under _keanu-examples_ directory. I did not add a specific logger config for those, but it could be easily done if we decided that we want to change logging settings for them.

### CODE REVIEW

File: keanu-examples/bayesianMethodsForHackers/src/test/java/com/examples/ChallengerDisasterTest.java [hunk 1]
```diff
@@ -3,12 +3,16 @@
...
 import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.within;
 
 public class ChallengerDisasterTest {
 
+    private final Logger log = LoggerFactory.getLogger(ChallengerDisasterTest.class);
```
Comment (migwellian) [inline keanu-examples/bayesianMethodsForHackers/src/test/java/com/examples/ChallengerDisasterTest.java]: The convention is always to make it `static`. However @gordoncaleb has just pointed out to me that you can annotate the class with Lombok's `@Slf4j` and then you can remove this line!

File: keanu-project/src/main/resources/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Root logger option
+log4j.rootLogger=INFO, stdout
+log4j.logger.io.improbable.keanu=INFO
+log4j.logger.org.nd4j=DEBUG
```
Comment (migwellian) [inline keanu-project/src/main/resources/log4j.properties]: shall we whack all of these up to WARN as the default?

File: keanu-project/src/test/resources/log4j.properties [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Root logger option
-log4j.rootLogger=WARN, stdout
-log4j.logger.io.improbable.keanu=DEBUG
-log4j.logger.org.nd4j=DEBUG
+log4j.rootLogger=ERROR
```
Comment (migwellian) [inline keanu-project/src/test/resources/log4j.properties]: why did you remove `, stdout`?

File: keanu-project/src/main/resources/log4j.properties (hunk 2 above)
Comment (ieva-improbable) [reply keanu-project/src/main/resources/log4j.properties]: Yeah, I think that would make sense, but wasn't sure what defaults others prefer.

Comment (ieva-improbable) [reply keanu-project/src/main/resources/log4j.properties]: We do have a bunch of logs that are on INFO level, these would not be displayed if we change it to WARN. Do you think that'd be fine?

File: keanu-project/src/test/resources/log4j.properties (hunk 3 above)
Comment (ieva-improbable) [reply keanu-project/src/test/resources/log4j.properties]: By accident, thanks for spotting!

File: keanu-project/src/main/resources/log4j.properties (hunk 2 above)
Comment (migwellian) [reply keanu-project/src/main/resources/log4j.properties]: I doubt anyone would miss them. Worth getting 1 or 2 more opinions though.

Comment (wesleyw72) [reply keanu-project/src/main/resources/log4j.properties]: I think WARN by default is fine. If anyone wants to particularly probe into whats happening they can override it.

**Comment by wesleyw72** [comment]
LGTM
